docs/cookbook/prompt_engineering/chaining_based/prompt_paraphrasing.md

```diff
@@ -0,0 +1,97 @@
+# Prompt Paraphrasing: Generating Diverse Prompts for LLM Testing and Evaluation
+
+[Prompt Paraphrasing](https://direct.mit.edu/tacl/article/doi/10.1162/tacl_a_00324/96460/How-Can-We-Know-What-Language-Models-Know) is not a prompt engineering technique, but rather a prompt generation technique used to create ensembles of prompts for testing or other prompt engineering techniques. In this example, we cover a specific method of generating prompts mentioned in the paper whereby a prompt is translated into $B$ versions in another language, then backtranslated into $B^2$ versions to English.
+
+??? tip "Mirascope Concepts Used"
+
+    - [Prompts](../../../learn/prompts.md)
+    - [Calls](../../../learn/calls.md)
+    - [Response Models](../../../learn/response_models.md)
+
+## Implementation
+
+Let's implement the Prompt Paraphrasing technique using Mirascope:
+
+```python
+from pydantic import BaseModel, Field
+
+from mirascope.core import openai, prompt_template
+
+
+class Translations(BaseModel):
+    translations: list[str] = Field(
+        ..., description="The list of translations into the requested language"
+    )
+
+
+@openai.call(model="gpt-4o-mini", response_model=Translations)
+@prompt_template(
+    """
+    For this phrase: {phrase}
+
+
+    Give {num_translations} translations in {language}
+    """
+)
+def translate(phrase: str, language: str, num_translations: int): ...
+
+
+def prompt_paraphrasing(query: str, num_translations: int = 3):
+    spanish_translations = translate(
+        phrase=query,
+        language="Spanish",
+        num_translations=num_translations,
+    )
+    # Avoid Duplicates
+    prompt_varations = set()
+    for spanish_phrase in spanish_translations.translations:
+        back_translations = translate(
+            spanish_phrase, language="English", num_translations=3
+        )
+        prompt_varations.update(back_translations.translations)
+    return prompt_varations
+
+
+print(
```

Comment:
again, get the response and print the content to be consistent with other recipes and sections

Reply:
this is a extraction that returns a set so other than typing the function not sure what else to do?